src/vinecop/tools_select.cpp

```diff
@@ -298,12 +298,81 @@ namespace tools_select {
+    
+            // iteratively search for an edge in lower tree that shares all 
+            // indices in the conditioned set + diagonal entry
+            for (size_t k = 1; k < t; ++k) {
+                auto reduced_set = cat(mat(t, col), ning_set);
+                for (auto e : boost::edges(trees_[t - k])) {
+                    if (is_same_set(trees_[t - k][e].all_indices, reduced_set)) {
+                        // next matrix entry is conditioned variable of new edge
+                        // that's not equal to the diagonal entry of this column
+                        auto e_new = trees_[t - k][e];
+                        ptrdiff_t pos = 
+                            find_position(mat(t, col), e_new.conditioned);
+                        mat(t - k - 1, col) = 
+                            e_new.conditioned[std::abs(1 - pos)];
+                        if (pos == 1) {
+                            e_new.pair_copula.flip();
+                        }
+                        // assign fitted pair copula to appropriate entry, see
+                        // vinecopulib::Vinecop::get_pair_copula().
+                        pair_copulas_[t - 1 - k][col] = e_new.pair_copula;
+    
+                        // start over with conditioned set of next edge
+                        ning_set = e_new.conditioning;
+    
+                        // remove edge (must not be reused in another column!)
+                        size_t v0 = boost::source(e, trees_[t - k]);
+                        size_t v1 = boost::target(e, trees_[t - k]);
+                        boost::remove_edge(v0, v1, trees_[t - k]);
+                        break;
+                    }
+                }
+            }
+        }
+    
+        // The last column contains a single element which must be different
+        // from all other diagonal elements. Based on the properties of an
+        // R-vine matrix, this must be the element next to it.
+        mat(0, d_ - 1) = mat(0, d_ - 2);
+    
+        // change to user-facing format
+        // (variable index starting at 1 instead of 0)
+        auto new_mat = mat;
+        for (size_t i = 0; i < d_; ++i) {
+            for (size_t j = 0; j < d_ - i; ++j) {
+                new_mat(i, j) += 1;    
+            }
+        }
+        vine_matrix_ = RVineMatrix(new_mat);
+    }
 
-    bool FamilySelector::belong_to_structure(size_t v0, size_t v1,
-                                             const VineTree& vine_tree,
-                                             const RVineMatrix& vine_matrix) {
-
-        bool res = false;
+    bool FamilySelector::belongs_to_structure(size_t v0, size_t v1,
+                                              const VineTree& vine_tree) {
+        
+        // -1 means no common neighbor in previous tree
         if (find_common_neighbor(v0, v1, vine_tree) > -1) {
             // conditioned sets
             auto conditioned0 = vine_tree[v0].conditioned;
```

Comment:
Wouldn't it be "nicer" to use all_indices rather than conditioned/conditioning here? We would avoid the additional `cat`...